api/helpers/airTable.js

```diff
@@ -62,31 +70,66 @@ function formatTime(timeInSeconds) {
   return dayjs().hour(0).minute(0).second(0).add(timeInSeconds, 'second').format('HH:mm');
 }
 
-async function getAllRecords(tableName, includeId = false) {
+async function getAllRecords(tableName, includeId = false, linkedFields) {
   try {
     const allRecordsRaw = await module.exports.client().table(tableName).select().all();
 
-    return allRecordsRaw.map((record) =>
-      includeId // if records don't already have a unique identifier column (e.g. events), it's useful to include the record ID from AirTable
-        ? {
-            id: record.id,
-            ...record.fields,
-          }
-        : record.fields,
+    return await Promise.all(
+      allRecordsRaw.map(async (record) => {
+        if (linkedFields?.length) {
+          record = await addLinkedFields(tableName, record, linkedFields);
+        }
+        return includeId // if records don't already have a unique identifier column (e.g. events), it's useful to include the record ID from AirTable
+          ? {
+              id: record.id,
+              ...record.fields,
+            }
+          : record.fields;
+      }),
     );
   } catch (error) {
     return error;
   }
 }
 
-async function getRecordById(tableName, recordId) {
+// Adding linked fields to a record
+// TODO: this needs a JSDoc
```

Comment:
Please can you add a JSDoc for this function (ideally) or if not, remove this comment

Reply:
Comment removed, thanks.